tests/ResponseTest.php

```diff
@@ -111,6 +112,64 @@ public function testSetContent()
         $this->assertSame('Hello World', (string) $response->getBody());
     }
 
+    public function testSetContentWithResource()
+    {
+        $response = new Response();
+
+        $this->assertSame($response, $response->setContent(fopen('php://input', 'rb')));
+        $this->assertInstanceOf(MessageBody::class, $response->getBody());
```

Comment:
You can actually check that it's a `MessageBodyResource` here.

Reply:
Please remove the `assertInstanceOf(MessageBody::class`, the next line now implies it!